src/passes/references/expectedLocationAnalyser.ts

```diff
@@ -108,29 +108,37 @@ export class ExpectedLocationAnalyser extends ASTMapper {
     }
 
     const parameterTypes = getParameterTypes(node, ast);
+    // When calling `push`, the function recieves two argument nonetheless the argument is just one
+    // This does not explode because javascript does not gives an index out of range exception
+    // It should be handled but not sure it is here
     parameterTypes.forEach((t, index) => {
+      const argI = node.vArguments[index];
       if (t instanceof PointerType) {
         if (node.kind === FunctionCallKind.StructConstructorCall) {
           // The components of a struct being assigned to a location are also being assigned to that location
           const expectedLocation = this.expectedLocations.get(node);
           if (expectedLocation !== undefined && expectedLocation !== DataLocation.Default) {
-            this.expectedLocations.set(node.vArguments[index], expectedLocation);
+            this.expectedLocations.set(argI, expectedLocation);
             return;
           }
 
           // If no expected location, check the type associated with the parent struct constructor
           const structType = getNodeType(node, ast.compilerVersion);
           assert(structType instanceof PointerType);
           if (structType.location !== DataLocation.Default) {
-            this.expectedLocations.set(node.vArguments[index], structType.location);
+            this.expectedLocations.set(argI, structType.location);
           } else {
             //Finally, default to the type in the pointer itself if we can't infer anything else
-            this.expectedLocations.set(node.vArguments[index], t.location);
+            this.expectedLocations.set(argI, t.location);
           }
+          // Need this case for when you call darray.push([1,2,3]) the tuple gets memory location
+        } else if (argI instanceof TupleExpression && argI.isInlineArray) {
+          this.expectedLocations.set(argI, DataLocation.Memory);
```

Comment:
I'm unconvinced that this is the correct solution in general as a tuple propagates its expected location, and the children should not have expected location memory, e.g. what if they were structs? Solving this should probably be a separate PR though